### PR TITLE
Add datasets endpoint

### PR DESCRIPTION
**Pull request by magnurud**

### CODE REVIEW

File: las/client.py [hunk 1]
```diff
@@ -553,23 +628,44 @@ def delete_documents(
...
+        response = self._make_request(requests.delete, '/documents', params=params)
+
+        if delete_all:
+            params['nextToken'] = response['nextToken']
+
+            while params['nextToken']:
+                intermediate_response = self._make_request(requests.delete, '/documents', params=params)
+                response['documents'].extend(intermediate_response.get('documents'))
+                params['nextToken'] = intermediate_response['nextToken']
+                logger.info(f'deleted {len(response["documents"])} so far')
```
Comment (augustak) [inline las/client.py:666]:
```suggestion
                logger.info(f'Deleted {len(response["documents"])} documents so far')
```

File: las/client.py [hunk 2]
```diff
@@ -588,27 +684,51 @@ def get_document(self, document_id: str) -> Dict:
         """
         return self._make_request(requests.get, f'/documents/{document_id}')
 
-    def update_document(self, document_id: str, ground_truth: Sequence[Dict[str, Union[Optional[str], bool]]]) -> Dict:
+    def update_document(
+        self,
+        document_id: str,
+        *,
```
Comment (augustak) [inline las/client.py:690]: This is potentially a breaking change. Users who have not used kwarg for ground_truth will get an error moving forward. Do we want this?

File: las/client.py [hunk 3]
```diff
@@ -813,6 +933,95 @@ def delete_model(self, model_id: str) -> Dict:
...
+            'maxResults': max_results,
+            'nextToken': next_token,
+        }
+        return self._make_request(requests.get, f'/models/{model_id}/dataBundles', params=params)
+
+    def update_data_bundle(
+        self,
+        model_id: str,
+        data_bundle_id: str,
+        **optional_args,
```
Comment (augustak) [inline las/client.py:989]: There is some inconsistencies in when * is used. Should we refactor this across the SDK and publish a new major version?

File: setup.py [hunk 4]
```diff
@@ -10,7 +10,7 @@
 
 setup(
     name='lucidtech-las',
-    version='3.3.1',
+    version='3.4.0',
```
Comment (augustak) [inline setup.py:13]: See comments above

File: tests/test_data_bundles.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    )
+    assert_data_bundle(response)
+
+
+def assert_data_bundle(response):
+    assert 'modelId' in response, 'Missing modelId in response'
+    assert 'dataBundleId' in response, 'Missing dataBundleId in response'
+    assert 'name' in response, 'Missing name in response'
+    assert 'description' in response, 'Missing description in response'
+
```
Comment (augustak) [inline tests/test_data_bundles.py:54]:
```suggestion
    assert 'description' in response, 'Missing description in response'
    assert 'createdTime' in response, 'Missing createdTime in response'
    assert 'status' in response, 'Missing status in response'

```

File: tests/test_datasets.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+@pytest.mark.parametrize('name_and_description', util.name_and_description_combinations(at_least_one=True))
+def test_update_dataset(client: Client, name_and_description):
+    response = client.update_dataset(service.create_dataset_id(), **name_and_description)
+    assert_dataset(response)
+
+
+def assert_dataset(response):
+    assert 'datasetId' in response, 'Missing datasetId in response'
+    assert 'name' in response, 'Missing name in response'
+    assert 'description' in response, 'Missing description in response'
```
Comment (augustak) [inline tests/test_datasets.py:49]:
```suggestion
    assert 'description' in response, 'Missing description in response'
    assert 'createdTime' in response, 'Missing createdTime in response'
    assert 'updatedTime' in response, 'Missing updatedTime in response'
    assert 'version' in response, 'Missing version in response'
```

**Review by augustak — CHANGES_REQUESTED** [review]

File: las/client.py [hunk 7]
```diff
@@ -553,23 +628,44 @@ def delete_documents(
...
         :param consent_id: Ids of the consents that marks the owner of the document
         :type consent_id: Optional[Queryparam]
+        :param dataset_id: Ids of the datasets to be deleted
+        :type dataset_id: Optional[Queryparam]
         :param max_results: Maximum number of documents that will be deleted
         :type max_results: Optional[int]
         :param next_token: A unique token for each page, use the returned token to retrieve the next page.
         :type next_token: Optional[str]
+        :param delete_all: delete all documents that match the given parameters, without worrying about next-token.
+        This is not possible if max_results is specified
```
[the remark this answers is not in the document]
Comment (magnurud) [reply las/client.py:638]: Will it have no effect? Or will we throw an error? I think it is better to throw an error if a user tries to specify both.

File: setup.py (hunk 4 above)
Comment (magnurud) [reply setup.py:13]: Maybe the breaking changes should be performed in a bigger refactoring?

File: las/client.py (hunk 2 above)
Comment (magnurud) [reply las/client.py:691]: no, we do not for now.

File: las/client.py (hunk 3 above)
Comment (magnurud) [reply las/client.py:989]: will fix in other PR

File: las/client.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (augustak) [reply las/client.py:638]: We can throw an error instead

**Review by augustak — APPROVED** [review]

**Pull request merged**